sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -507,6 +509,136 @@ case class GpuArrayMax(child: Expression) extends GpuUnaryExpression with Implic
     }
 }
 
+case class GpuArraysZip(children: Seq[Expression]) extends GpuExpression with ExpectsInputTypes {
+
+  override def inputTypes: Seq[AbstractDataType] = Seq.fill(children.length)(ArrayType)
+
+  @transient override lazy val dataType: DataType = {
+    val fields = children.zip(arrayElementTypes).zipWithIndex.map {
+      case ((expr: NamedExpression, elementType), _) =>
+        StructField(expr.name, elementType, nullable = true)
+      case ((_, elementType), idx) =>
+        StructField(idx.toString, elementType, nullable = true)
+    }
+    ArrayType(StructType(fields), containsNull = false)
+  }
+
+  override def nullable: Boolean = children.exists(_.nullable)
+
+  @transient private lazy val arrayElementTypes =
+    children.map(_.dataType.asInstanceOf[ArrayType].elementType)
+
+  override def withNewChildrenInternal(newChildren: IndexedSeq[Expression]): Expression =
+    legacyWithNewChildren(newChildren)
+
+  override def columnarEval(batch: ColumnarBatch): Any = {
+    val numRows = batch.numRows()
+    val numCols = children.size
+    // A container collects all column vectors in use, to ease the release on exceptions.
+    val workBuffers = mutable.ArrayBuffer[ColumnVector]()
+
+    if (numCols == 0) {
+      return GpuScalar(new GenericArrayData(Array.empty[Any]), dataType)
+    }
+
+    // Prepare input columns
+    closeOnExcept(workBuffers) { _ =>
+      children.foreach { expr =>
+        workBuffers += GpuExpressionsUtils.columnarEvalToColumn(expr, batch).getBase
+      }
+    }
+    val inputs = workBuffers.toArray
+    workBuffers.clear()
+
+    withResource(inputs) { inputs =>
+      // Compute array sizes of input arrays
+      closeOnExcept(workBuffers) { _ =>
+        (0 until numCols).foreach { i =>
+          workBuffers += inputs(i).countElements()
+        }
+      }
+
+      // Pick max array size of each row.
+      // Replace with zero if there exists null among input values.
+      // [1, 3, 5, null, 4]
+      // [2, 4, 3,    8, 0]   => [4, 4, 5, 0, 7]
+      // [4, 2, 3,   10, 7]
+      val maxSize = closeOnExcept(workBuffers) { _ =>
+        val max = withResource(ColumnVector.makeList(workBuffers: _*)) { sizes =>
+          workBuffers.foreach(_.close())
+          workBuffers.clear()
+          sizes.listReduce(SegmentedReductionAggregation.max())
```

Comment:
Is it faster to create a list and then do a `listReduce`? or would it be faster to just keep a running NULL_MAX of the columns as you go?  The running max will use less memory, so I am inclined to go with that over this implementation, but it would definitely be more complicated to write, so I was curious what others thought about this.

Reply:
I am not sure which approach is faster. I chose the `listReduce` over max iterations because it invokes less times of native kernel calls. Perhaps we need a mini benchmark to provide some empirical evidence ?

Reply:
I think this is fine as is. This is working, so there is less of a reason to change it.